python/mxnet/__init__.py

```diff
@@ -106,3 +110,13 @@
 from . import _api_internal
 from . import api
 from . import container
+
+# Clean subprocesses when MXNet is interrupted
+def mxnet_excepthook(exctype, value, trbk):
+    print('\n'.join(traceback.format_exception(exctype, value, trbk)))
+    if hasattr(multiprocessing, 'active_children'):
+        # pylint: disable=not-callable
+        for p in multiprocessing.active_children():
+            p.terminate()
+
+sys.excepthook = mxnet_excepthook
```

Comment:
This modifies global Python state and we have no guarantee that the user (or one of his libraries) has overwritten this prior to our modification. Why not use the `p.daemon=True` setting for subprocesses started by MXNet instead?

> When a process exits, it attempts to terminate all of its daemonic child processes.

https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Process.daemon